tests/neg-macros/i7839.check

```diff
@@ -0,0 +1,8 @@
+-- Error: tests/neg-macros/i7839.scala:4:18 ----------------------------------------------------------------------------
+4 |  ${ transformImpl } // error
+  |                  ^
+  |                  Malformed macro parameter: dummyImplicit
+  |
+  |                  Parameters may only be:
```

Comment:
Why does it resolve `DummyImplicit` at all if it is neither of the two cases below?

Reply:
This is a static check that happens after typechecking. We do not modify in any way the typing of expressions within top level splices, we only restrict it based terms that can be interpreted when evaluating the macro.